### PR TITLE
OCPBUGSM-46219: OCPBUGSM-46220: Update golang version to 1.18.1.

### PR DESCRIPTION
**Pull request by eliorerz**
Due to several CVE bugs we need to update golang version to >= v1.18.1.
https://bugzilla.redhat.com/show_bug.cgi?id=2077688
https://bugzilla.redhat.com/show_bug.cgi?id=2077689

This PR updates golang version to v1.18.1 and golangci to v1.46.0.

/cc @omertuc

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift/assisted-installer-agent
 
-go 1.17
+go 1.18
```
Comment (omertuc) [inline go.mod:3]: Same concern as 

https://github.com/openshift/assisted-installer/pull/581#discussion_r1027328537

**Review by gamli75 — APPROVED** [review]

**Pull request merged**